### PR TITLE
chore: ensure release notes always come from Clerk

### PR DESCRIPTION
**Pull request by ckerr**
As discussed in last week's WG Releases meeting in the 9.0.0 postmortem. Now with tests.

CC @electron/wg-releases 

#### Description of Change

* Ensure that the release notes generator always pulls text from Clerk's PR comments.
* Allow the release notes cache folder to be specified by environment variable
* Add tests

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,6 +53,7 @@
     "request": "^2.88.0",
     "semver": "^5.6.0",
     "shx": "^0.3.2",
+    "sinon": "^9.0.1",
```
Comment (jkleinsc) [inline package.json:56]: These changes should be in spec-main/package.json

File: yarn.lock [hunk 2]
```diff
@@ -135,6 +135,42 @@
   dependencies:
     any-observable "^0.3.0"
 
+"@sinonjs/commons@^1", "@sinonjs/commons@^1.6.0", "@sinonjs/commons@^1.7.0", "@sinonjs/commons@^1.7.2":
```
Comment (jkleinsc) [inline yarn.lock:138]: These changes should be in spec-main/yarn.lock

File: package.json (hunk 1 above)
Comment (ckerr) [reply package.json]: I've moved the sinon dependencies into spec-main/ in 839eebe12f4721c6f0d3e2f5f2bfefe1b8dce9eb

I'm a little unclear on the division of dependencies? The change you suggested is what makes intuitive sense to me, but I don't undersatnd why the top-level package.json is where the dependencies on @types/chai, @types/chai-as-promised, @types/dirty-chai, @types/mocha, eslint-plugin-mocha are specified...?

File: yarn.lock (hunk 2 above)
Comment (ckerr) [reply yarn.lock]: Moved in 839eebe

File: package.json [hunk 3]
```diff
@@ -142,4 +142,4 @@
   "dependencies": {
     "@types/temp": "^0.8.34"
   }
-}
```
Comment (jkleinsc) [inline package.json:145]: This change shouldn't be committed

File: package.json (hunk 1 above)
Comment (jkleinsc) [reply package.json]: The way I see it is that the dependencies should be located where they are used.  If we are using them in multiple places then it makes sense to put them at the top level.

**Pull request merged**